python/ray/experimental/sgd/pytorch/utils.py

```diff
@@ -4,32 +4,84 @@
 
 from ray.experimental.sgd.utils import TimerStat
 
-
-def train(model, train_iterator, criterion, optimizer, config):
-    """Runs 1 training epoch"""
+amp = None
+
+try:
```

Comment:
Maybe leave a comment on why this is optional/what doesn't work if it isn't present?

Reply:
done, good idea